src/include/duckdb/common/preserved_error.hpp

```diff
@@ -39,6 +39,9 @@ class PreservedError {
 	//! Let's us do things like 'if (error)'
 	DUCKDB_API operator bool() const;
 	DUCKDB_API bool operator==(const PreservedError &other) const;
+	DUCKDB_API shared_ptr<Exception> GetError() {
```

Comment:
Could this instead return a reference to the Exception, e.g. `const shared_ptr<Exception> &`